apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.RetryConformanceTests/RetryConformanceTest.cs

```diff
@@ -81,49 +81,34 @@ public async Task RetryTest(RetryTest test)
     /// <summary>
     /// Handle full flow of each individual test case from resource creation to running the testing, verify if it passed or failed and delete resource post usage.
     /// </summary>
-    private async Task RunTestCase(InstructionList instructionList, Method method, bool expectSuccess, bool specifyPreconditions)
+    private async Task RunTestCaseAsync(InstructionList instructionList, Method method, bool expectSuccess, bool specifyPreconditions)
     {
         var context = CreateTestContext(method);
-        var response = await CreateRetryTestResource(instructionList, method);
-        try
+        var response = await CreateRetryTestResourceAsync(instructionList, method);
+
+        if (expectSuccess)
         {
-            if (expectSuccess)
+            RunRetryTest(response, context, method.Group, specifyPreconditions);
+            var postTestResponse = await GetRetryTestAsync(response.Id);
+            Assert.True(postTestResponse.Completed, "Expected retry test completed to be true, but was false.");
+        }
+        else
+        {
+            try
             {
                 RunRetryTest(response, context, method.Group, specifyPreconditions);
-                var postTestResponse = await GetRetryTest(response.Id);
-                Assert.True(postTestResponse.Completed, "Expected retry test completed to be true, but was false.");
-            }
-            else
-            {
-                try
-                {
-                    RunRetryTest(response, context, method.Group, specifyPreconditions);
 
-                    // storage.buckets.setIamPolicy has no preconditions implemented in .NET and hence will retry successfully
-                    if (!method.Name.Contains("buckets.setIamPolicy"))
-                    {
-                        Assert.False(response.Completed);
-                    }
-                }
-                catch (GoogleApiException ex) when (InstructionContainsErrorCode(ex.HttpStatusCode))
+                // storage.buckets.setIamPolicy has no preconditions implemented in .NET and hence will retry successfully
+                if (!method.Name.Contains("buckets.setIamPolicy"))
```

Comment:
Can you move this case to inside the if? See my comment there.

Reply:
Same comment as above